scripts/area_plot.py

```diff
@@ -0,0 +1,613 @@
+                                           )
+    # Gets the colormap 
+    cmap = define_join_cmap(upper_)
+    upper_.index = upper_.index.droplevel('rough')
+    lower_.index = lower_.index.droplevel('rough')
+
+    # Plots the data
+    fig_ = plot_area(upper_.astype(float), lower_.astype(float), cmap)
+
+    return fig_
+
+
+# Sets up the main arguments for argparse.
+parser_one = argparse.ArgumentParser(
+    description=('A set of functions to generate diagnostic stacked area '
+                 'plots from metagenomic outputs.'),
+    prog=('area_plotter'),
+    )
+parser_one.add_argument('-t', '--table', 
+                    help=('The abundance table as a tsv classic biom '
+                          '(features as rows, samples as columns) containing'
+                          'absloute or relative abundance for the samples.'),
+                    required=True,
+                    )
+parser_one.add_argument('-f', '--figure',
+                    help=('The location for the final figure'),
+                    required=True,
+                    )
+parser_one.add_argument('-s', '--samples', 
+                    help=('A text file with the list of samples to be include' 
+                         ' (one per line). If no list is provided, then data '
+                         'from all columns in the table (except the one '
+                         'specifying taxonomy) will be used.'),
+                    )
+parser_one.add_argument('--mode', choices=(["metaphlan", "kracken", "marker"]),
+                    help=('The software generating the table to make parsing'
+                          'easier. Optens are kracken, metaphlan, marker '
+                          '(i.e. CTMR amplicon) and other for all other '
+                          'processing types.'),
+                    default='kracken'
+                    )
+parser_one.add_argument('-l', '--level',
+                    help=('The taxonomic level (as an integer) to plot the '
+                          'data.'),
+                    default=3,
+                    type=int,
+                    )
+parser_one.add_argument('--abund-thresh',
+                    help=("the minimum abundance required to display a group."),
+                    default=0.01,
+                    type=float,
+                    )
+parser_one.add_argument('--group-thresh',
+                    help=("The maximum number of groups to be displayed in "
+                        "the graph."),
+                    default=8,
+                    type=int,
+                    )
+parser_one.add_argument('-c', '--colormap',
+                    help=("The qualitative colormap to use to generate your "
+                          "plot. Refer to colorbrewer for options. If a "
+                          "selected colormap exceeds the number of groups "
+                          "(`--group-thresh`) possible, it will default to "
+                          "Set3."),
```

Comment:
Remember to also replace this default with something like `[%(default)]` like I showed above, to reduce duplication and having to update it in two places if we change the default.

Reply:
This isn't a case where where it's the default per the argparse. It's a warning about a real world boundary: there is no other viable option and it's hard coded that way for a reason. There are only two qualitative colormaps in color brewer which allow you to display more than 9 colors. Changing the default colormap here will not change the fact that if you exceed the allowed colors of a colormap, it *should* default to something that allows more colors. 

So, the default here could be set to something like "Set1" which has 8 colors, but if you was to display 12 colors, you default to "Set3" regardless of the argparse default.